src/GdbCommandHandler.cc

```diff
@@ -44,8 +44,18 @@ import re
 def gdb_unescape(string):
     result = ""
     pos = 0
-    while pos < len(string):
-        result += chr(int(string[pos:pos+2], 16))
+    len = len(string)
+    # care for unexpected string length
+    if len % 2:
+        len -= 1
+        result = "!"
```

Comment:
Can an odd length actually happen?
If it can happen why is ignoring the last character the right thing to do?

Reply:
as noted in the related issues there are different reasons for invalid packages and rr should not abort on these, neither in C++ (normally completely aborted process because of unhandled exception) nor in python (aborting the call with a stacktrace)